components/user/CourseCard.tsx

```diff
@@ -20,13 +19,16 @@ const CourseCard = ({ code }: { code: string }) => {
   const course = data.result
 
   return (
-    <Link href={`/courses/${code}`} className='flex flex-row gap-x-4 '>
+    <Link
+      href={`/courses/${code}`.toLowerCase()}
+      className='flex flex-row gap-x-4 '
+    >
       <div className='w-60 border rounded-lg p-2 px-4 flex-shrink-0 snap-start'>
-        <p className='font-semibold'>{format(course.code)}</p>
-        <p className=''>{course.name}</p>
+        <p className='font-semibold'>{format(course?.code)}</p>
+        <p className=''>{course?.name}</p>
```

Comment:
Can you explain to me why you added the optional chaining?

Reply:
I was testing something for the same reasons mentioned above. All works now without it. I will remove it now.